Duplicati/UnitTest/ProblematicPathTests.cs

```diff
@@ -21,6 +22,153 @@ private static void WriteFile(string path, byte[] contents)
             }
         }
 
+        [Test]
+        [Category("ProblematicPath")]
+        public void DirectoriesWithWildcards()
+        {
+            const string file = "file";
+            List<string> directories = new List<string>();
+
+            // Keep expected match counts since they'll differ between
+            // Linux and Windows.
+            var questionMarkWildcardShouldMatchCount = 0;
+            var verbatimAsteriskShouldMatchCount = 0;
+
+            const string asterisk = "*";
+            string dirWithAsterisk = Path.Combine(this.DATAFOLDER, asterisk);
+            // Windows does not support literal asterisks in paths.
+            if (!Platform.IsClientWindows)
+            {
+                SystemIO.IO_OS.DirectoryCreate(dirWithAsterisk);
+                WriteFile(SystemIO.IO_OS.PathCombine(dirWithAsterisk, file), new byte[] {0});
+                directories.Add(dirWithAsterisk);
+                questionMarkWildcardShouldMatchCount++;
+                verbatimAsteriskShouldMatchCount++;
+            }
+
+            const string questionMark = "?";
+            string dirWithQuestionMark = Path.Combine(this.DATAFOLDER, questionMark);
+            // Windows does not support literal question marks in paths.
+            if (!Platform.IsClientWindows)
+            {
+                SystemIO.IO_OS.DirectoryCreate(dirWithQuestionMark);
+                WriteFile(SystemIO.IO_OS.PathCombine(dirWithQuestionMark, file), new byte[] { 1 });
+                directories.Add(dirWithQuestionMark);
+                questionMarkWildcardShouldMatchCount++;
+            }
+
+            // Include at least one single character directory in Windows
+            // for a '?' wildcard can match on
+            const string singleCharacterDir = "X";
+            string dirWithSingleCharacter = Path.Combine(this.DATAFOLDER, singleCharacterDir);
+            SystemIO.IO_OS.DirectoryCreate(dirWithSingleCharacter);
+            WriteFile(SystemIO.IO_OS.PathCombine(dirWithSingleCharacter, file), new byte[] { 1 });

```

Comment:
Can we make the file contents here different than the others?  This way, we can hopefully detect a case where the restore operation restores the wrong file.

Reply:
Fixed in latest commit.